docs/rfcs/env-vars.md

```diff
@@ -0,0 +1,216 @@
+any-typed variable and the original representation as a string is stored
+for numeric types. The `yaml.v3` parser mostly follows the YAML v1.2
+specification with [*some
+exceptions*](https://github.com/go-yaml/yaml#compatibility). You can see
+how it works for some edge cases in this example:
+[*https://go.dev/play/p/RtPmH8aZA1X*](https://go.dev/play/p/RtPmH8aZA1X).
+
+When unmarshalling, we use mapstructure with WeaklyTypedInput
+**disabled**. We check via a hook an `AsString` method from confmap.Conf
+and use its return value when it is valid and we are mapping to a string
+field. This method has default casting rules for unambiguous scalar
+types but may return the original representation depending on the
+construction of confmap.Conf
+
+For using this notation in inline mode (e.g.`http://endpoint/${env:PATH}`), we
+use the `AsString` method from confmap.Conf.
+
+### Character set
+
+An environment variable identifier must be a nonempty ASCII alphanumeric
+or underscore starting with an alphabetic or underscore character. Its
+maximum length is 200 characters. Both syntaxes support recursive
+resolution.
+
+When an invalid identifier is found, a warning log is emitted, and the
+string is kept as-is.
+
+### Comparison table with current behavior
+
+This is a comparison between the current and desired behavior for
+loading a field with the braces syntax, `env` syntax.
+
+| Raw value    | Field type | Current behavior, `${ENV}`, single field | Current behavior,  `${env:ENV}` , single field | Desired behavior, entire field | Desired behavior, inline string field |
+|--------------|------------|------------------------------------------|------------------------------------------------|--------------------------------|---------------------------------------|
+| `123`        | integer    | 123                                      | 123                                            | 123                            | n/a                                   |
+| `0123`       | integer    | 83                                       | 83                                             | 83                             | n/a                                   |
+| `0123`       | string     | 0123                                     | 83                                             | 0123                           | 0123                                  |
+| `0xdeadbeef` | string     | 0xdeadbeef                               | 3735928559                                     | 0xdeadbeef                     | 0xdeadbeef                            |
+| `"0123"`     | string     | "0123"                                   | 0123                                           | 0123                           | 0123                                  |
+| `!!str 0123` | string     | !!str 0123                               | 0123                                           | 0123                           | 0123                                  |
```

Comment:
For 'Desired behavior, inline string field' here, the behavior of the File Configuration SDK would be to set `!!str 0123`. I think this is a rare edge case, but as written it would set `0123` since the `AsString` method would not return the original representation

Reply:
>For 'Desired behavior, inline string field' here, the behavior of the File Configuration SDK would be to set `!!str 0123`.

Can you link to where this is defined? I can't see where this is addressed this outside of declaring a suggested minimum version of the YAML spec.